src/plugins/footnotes/index.ts

```diff
@@ -227,6 +294,50 @@ export default () => {
               )
             }
           }
+          if (can.editArticle) {
```

Comment:
I've just noticed that this piece of code is inside `state.doc.descendants`. This means that the widget is being added to **all the tables** for the general note, not just the selected one. I see that you're hiding the button and then showing it on selection, which is why it appears to be working fine.

While this seems acceptable, it's worth noting that, this code should be outside of `state.doc.descendants` and should only add the widget to the selected node.

We can address this refinement later. I just wanted to bring it to your attention.